scene/main/scene_tree.cpp

```diff
@@ -1416,7 +1418,9 @@ SceneTree::SceneTree() {
 		if (env_path != String()) {
 			Ref<Environment> env = ResourceLoader::load(env_path);
 			if (env.is_valid()) {
+#ifndef _3D_DISABLED
```

Comment:
The whole `env_path` stuff should likely be compiled out instead of just this call (it breaks the logic).

Reply:
Are you sure? I thought the environment was for both 2D and 3D.

Reply:
I haven't looked at details, but yes `Environment` can be used in 2D too.

But this specific logic seems to load the configured default environment if there's one, and then if it's valid, sets it as fallback for World3D, otherwise it clears the setting or tells the user about it. So it seems this part only cares about 3D (i.e. `rendering/environment/defaults/default_environment` doesn't impact 2D).